service/src/main/kotlin/io/provenance/explorer/domain/entities/Transactions.kt

```diff
@@ -284,8 +285,11 @@ class TxMessageRecord(id: EntityID<Int>) : IntEntity(id) {
                         it[this.txMessage] = message
                         it[this.txMessageHash] = message.value.toDbHash()
                     }
+                    msgId = this.id
                 }
             }
+            return msgId
+        }
```

Comment:
There is a function on the table object called `insertAndGetId()` which will return the id. 

Would be cleaner to do it like this:
```
fun insert(blockHeight: Int, txHash: String, txId: EntityID<Int>, message: Any, type: String, module: String) =
            transaction {
                TxMessageTypeRecord.insert(type, module, message.typeUrl).let { typeId ->
                    findByTxHashAndMessageHash(txId.value, message.value.toDbHash())?.id
                        ?: TxMessageTable.insertAndGetId {
                            it[this.blockHeight] = blockHeight
                            it[this.txHash] = txHash
                            it[this.txHashId] = txId
                            it[this.txMessageType] = typeId
                            it[this.txMessage] = message
                            it[this.txMessageHash] = message.value.toDbHash()
                        }
                }
            }
```

This would return the `EntityID<Int>`, which you can extract the value as `this.value`.

Reply:
Oh sweet, that makes things easier.